### ISSUE TITLE
Failed hash check

### ISSUE THREAD

**Issue by SuperiorJT** [sodiumoxide/sodiumoxide#326]
I received this error when building on mingw64. I had to enable SODIUM_USE_PKG_CONFIG in order to successfully build. I checked the releases at https://download.libsodium.org/libsodium/releases/ and it looks like 1.0.17 was modified on 03-Apr-2019 so I think the checksum just needs to be updated. I don't have any experience in hash checking, so I might be wrong.

**Comment by kpp** [comment]
> https://github.com/libp2p/rust-libp2p/blob/2c66f82b110dca56eeca7fcaa0273f09f7e5fb56/protocols/noise/Cargo.toml#L28

@libp2p sorry guys your build is broken due to https://github.com/jedisct1/libsodium/issues/813. We will fix it ASAP. Would you please tell your preferences about releasing binaries into that issue?

**Comment by Erk-** [comment]
Are you planning to release a minor release with this fixed?

**Comment by kpp** [comment]
We are working on it

**Comment by kurnevsky** [comment]
And again... https://download.libsodium.org/libsodium/releases/

**Comment by kpp** [comment]
Oh no...

**Comment by kurnevsky** [comment]
Hm... Old libraries are back.

**Comment by mbelop** [comment]
Why don't you bundle it up? Would have solved a whole bunch of issues and remove the dependency on http_req (which I have to remove manually in my branch).

**Comment by kpp** [comment]
How?

**Comment by mbelop** [comment]
How what? How to bundle it?

**Comment by kpp** [comment]
You can’t do that for all binaries because the size of a crate should be not bigger than 20Mb

**Comment by dnaq** [comment]
Couldn’t we bundle the source?

**Comment by mbelop** [comment]
I only meant to propose bundling the source code.

**Comment by kpp** [comment]
Yeah but how would you build it?

**Comment by mbelop** [comment]
The same way it's built right now. You can take a look at how I did it: https://github.com/mbelop/sodiumoxide/commit/3e80b0d8972f7f83f836b82914aaea7409156738
But as I've said, I wanted to have only minimal changes to the existing build infrastructure.

**Comment by kpp** [comment]
Take a look at https://github.com/sodiumoxide/sodiumoxide/pull/300

I wanted to build the lib from sources but I failed to do this on windows

**Comment by sopium** [comment]
> You can’t do that for all binaries because the size of a crate should be not bigger than 20Mb

Maybe we can.

For MSVC, we only need `x64\Release\v140\static\libsodium.lib` and `Win32\Release\v140\static\libsodium.lib`. The two files are 973KiB compressed.

The mingw tarball is 985KiB compressed. If we only include `.a` files it's even smaller.

So that's 1.91MiB in total.

**Comment by kpp** [comment]
Huh. Sounds like a solution.

**Comment by kpp** [comment]
@SuperiorJT would you please check https://github.com/sodiumoxide/sodiumoxide/tree/bundled_libsodium branch ?

**Comment by kpp** [comment]
@sodiumoxide/owners can anyone check it on mingw64?

**Comment by Dylan-DPC** [comment]
@kpp you could add it to the CI though it will slow down the builds but that's fine

**Comment by kpp** [comment]
I will try

**Comment by dnaq** [comment]
I’ve heard some talk about azure pipes being a lot faster than Travis.
Might be a reason to switch.

**Comment by sainthkh** [comment]
I found the same issue compiling in native environment. 

But the difference is that the result of hash value changes every time I compile it. (I'm guessing that the header values are included when calculating hash. Or I might be terribly wrong.)

Here's the error message you can see: 

```
error: failed to run custom build command for `libsodium-sys v0.2.1`
process didn't exit successfully: `D:\projects\teatime-korean\packages\gateway\target\debug\build\libsodium-sys-f88bc9205f31dd6c\build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-env-changed=SODIUM_LIB_DIR
cargo:rerun-if-env-changed=SODIUM_SHARED
cargo:rerun-if-env-changed=SODIUM_USE_PKG_CONFIG
cargo:rerun-if-env-changed=VCPKGRS_DYNAMIC

--- stderr
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `"f0f32ad8ebd76eee99bb039f843f583f2babca5288a8c26a7261db9694c11467"`,
 right: `"d0ccc129253d0d51f09f8d030129041eb56fc3f488a0206babff2ef0b1752368"`:

Downloaded libsodium file failed hash check.

', C:\Users\faith\.cargo\registry\src\github.com-1ecc6299db9ec823\libsodium-sys-0.2.1\build.rs:162:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

Currently, only workaround is to use [vcpkg](https://github.com/Microsoft/vcpkg). 

For those who came here for workaround in PowerShell, do these:

1. Install [vcpkg](https://github.com/Microsoft/vcpkg).
2. Use user-wide integration with `.\vcpkg integrate install`.
3. Set env variables 
```
PS> $env:SODIUM_USE_PKG_CONFIG=1
PS> $env:VCPKGRS_DYNAMIC=1
```

If you're cmd user, use the code on the README.md page. 

Windows Version: Windows 10 Pro 1803
libsodium-sys version: 0.2.1
Visual Studio Version: 15.9.11

**Comment by kpp** [comment]
@sainthkh you can try `master` branch

**Comment by SuperiorJT** [comment]
0.2.2 is working for me

**Comment by kpp** [comment]
Resolved in 0.2.2
